node/bft/ledger-service/src/ledger.rs

```diff
@@ -41,12 +42,15 @@ use std::{
 
 /// The capacity of the LRU holding the recently queried committees.
 const COMMITTEE_CACHE_SIZE: usize = 16;
+/// The capacity of the cache holding the highest blocks.
+const BLOCK_CACHE_SIZE: usize = 10;
```

Comment:
for such a small cache I think something like a `CricularQueue` might be faster to query (even with linear search) and iterate, and pushing beyond capacity would automatically drop the oldest entry

Reply:
I'd rather not add a dependency for such a small performance gain.

Just in case you don't have to do too much research, and in case you like it, think a `VecDeque` would be worth it for similar reasons?

Reply:
it could be, yes

Reply:
Benchmarked BTreeMap locally, does not seem worth it to refactor:
- Elapsed time 10_000 iterations of 100MB Get: 84.875µs
- Elapsed time 10_000 iterations of 100MB Insert: 136.667µs

Reply:
Definitely appreciate the recommendation and learning about the datastructure trade-offs

Reply:
sounds good, if it's not a hot path we can probably spare several microseconds here :smiley: